api/src/test/java/uk/gov/dhsc/htbhf/card/CardServicesIntegrationTest.java

```diff
@@ -0,0 +1,102 @@
+package uk.gov.dhsc.htbhf.card;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.dhsc.htbhf.card.model.*;
+import uk.gov.dhsc.htbhf.errorhandler.ErrorResponse;
+
+import java.net.URI;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verify;
+import static org.springframework.http.HttpStatus.CREATED;
+import static org.springframework.http.HttpStatus.OK;
+import static uk.gov.dhsc.htbhf.assertions.IntegrationTestAssertions.assertValidationErrorInResponse;
+import static uk.gov.dhsc.htbhf.card.testhelper.CardRequestDTOTestDataFactory.aCardRequestWithAddress;
+import static uk.gov.dhsc.htbhf.card.testhelper.CardRequestDTOTestDataFactory.aValidCardRequest;
+import static uk.gov.dhsc.htbhf.card.testhelper.DepositFundsRequestDTOTestDataFactory.aDepositFundsRequestWithAmount;
+import static uk.gov.dhsc.htbhf.card.testhelper.DepositFundsRequestDTOTestDataFactory.aValidDepositFundsRequest;
+import static uk.gov.dhsc.htbhf.card.testhelper.DepositFundsResponseTestDataFactory.aValidDepositFundsResponse;
+import static uk.gov.dhsc.htbhf.card.testhelper.TestConstants.DEPOSIT_REFERENCE_ID;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class CardServicesIntegrationTest {
+
+    private static final URI ENDPOINT = URI.create("/v1/cards");
+    private static final String BALANCE_URL = ENDPOINT + "/1/balance";
+    private static final String DEPOSIT_URL = ENDPOINT + "/1/deposit";
+
+    private static final String CARD_SERVICES_DEPOSIT_URL = "http://localhost:8120/v1/cards/1/deposit";
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @MockBean
+    private RestTemplate restTemplateWithIdHeaders;
```

Comment:
Shouldn't an integration test be using wiremock rather than mocking the RestTemplate?

Reply:
That's what we're using for all our Integration tests, so I just did the same as the other services.